app/services/bulkrax/factory_class_finder.rb

```diff
@@ -2,28 +2,62 @@
 
 module Bulkrax
   class FactoryClassFinder
+    ##
+    # The v6.0.0 default coercer.  Responsible for converting a factory class name to a constant.
+    module DefaultCoercer
+      ##
+      # @param name [String]
+      # @return [Class] when the name is a coercible constant.
+      # @raise [NameError] when the name is not coercible to a constant.
+      def self.call(name)
+        name.constantize
+      end
+    end
+
+    ##
+    # A name coercer that favors classes that end with "Resource" but will attempt to fallback to
+    # those that don't.
+    module ValkyrieMigrationCoercer
+      SUFFIX = "Resource"
+
+      ##
+      # @param name [String]
+      # @param suffix [String] the suffix we use for a naming convention.
+      #
+      # @return [Class] when the name is a coercible constant.
+      # @raise [NameError] when the name is not coercible to a constant.
+      def self.call(name, suffix: SUFFIX)
+        if name.end_with?(suffix)
+          name.constantize
+        else
+          begin
+            "#{name}#{suffix}".constantize
```

Comment:
So this makes "Resource" a 'reserved' term referring only to a model which has a non-resource class as well?

I think this is the behavior in our hyrax adapter, but I don't think we perpetuated it outside of the adapter, did we? Should this behavior be in an adapter rather than this module? (just a suggestion/question, as I haven't looked into the code in Hyrax.... I just saw the behavior)

Reply:
It looks below like the coercer is configurable, so I guess someone could write their own... so maybe never mind, as I hadn't looked all the way down. lol